TrafficSimulator/Assets/TrafficLights/TrafficLightController.cs

```diff
@@ -0,0 +1,121 @@
+using UnityEngine;
+
+enum Mode{FIRST, SECOND};
+
+public class TrafficLightController : MonoBehaviour
```

Comment:
This logic needs to be remade. There are several loopholes that causes the controller to malfunction. Instead you should use the two pairs as the two different "directions", and in the logic simply loop through the first pair and set everything to green lights, and then the second pair and set everything to red lights.

For the second mode you simply switch which pair you set to green and set the other to red. There is no need for different logic depending on how many traffic lights there are.